types/react-select/src/types.d.ts

```diff
@@ -64,25 +64,54 @@ export interface CommonProps<
     options: OptionsType<OptionType>;
     selectOption: (option: OptionType) => void;
     selectProps: SelectProps<OptionType, IsMulti, GroupType>;
-    setValue: (value: ValueType<OptionType, IsMulti>, action: ActionTypes) => void;
+    setValue: (newValue: ValueType<OptionType, IsMulti>, action: SetValueAction, option: OptionType) => void;
```

Comment:
```suggestion
    setValue: (newValue: ValueType<OptionType, IsMulti>, action: SetValueAction, option?: OptionType) => void;
```

Should this be optional?

Reply:
You don't have to use all the parameters (i.e., `setValue` can be a function that is just `(newValue: ValueType<OptionType, IsMulti>, action: SetValueAction) => void`), but I believe the `option` parameter will always be specified. Can you provide an example where this is not the case?

Reply:
We have a few uses where `setValue` is called programmatically via `CommonProps`. Here's a contrived example:

```tsx
import Select, { components } from 'react-select';

const options = [
  { value: 'chocolate', label: 'Chocolate' },
  { value: 'strawberry', label: 'Strawberry' },
  { value: 'vanilla', label: 'Vanilla' },
];

export default function ExampleSelect() {
  return (
    <Select
      components={{
        Menu({ children, ...props }) {
          return (
            <components.Menu {...props}>
              <>
                <button
                  onClick={() => props.setValue(props.options, 'select-option')}
                >
                  Select all
                </button>
                {children}
              </>
            </components.Menu>
          );
        },
      }}
      options={options}
      isMulti
    />
  );
}
```

Although, now that I'm reviewing this logic, I believe we could programmatically set these values outside.

Reply:
Yes, thanks, after reviewing it does look there is precedent for it being optional (since it's not getting passed [here](https://github.com/JedWatson/react-select/blob/master/packages/react-select/src/Select.js#L832)).  I've made https://github.com/DefinitelyTyped/DefinitelyTyped/pull/50901 to fix the oversight.

Reply:
Thanks!